src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorTagHelperBinderPhase.cs

```diff
@@ -399,21 +402,23 @@ internal static bool IsTagHelperFromMangledClass(TagHelperDescriptor tagHelper)
                 {
                     // If this is a child content tag helper, we want to look at it's original type.
                     // E.g, if the type name is `Test.__generated__MyComponent.ChildContent`, we want to look at `Test.__generated__MyComponent`.
-                    TrySplitNamespaceAndType(typeName, out typeName, out var _);
+                    TrySplitNamespaceAndType(typeName, out var typeNameTextSpan, out var _);
+                    typeName = GetTextSpanContent(typeNameTextSpan, typeName);
                 }
-                if (!TrySplitNamespaceAndType(typeName, out var _, out var className))
+                if (!TrySplitNamespaceAndType(typeName, out var _, out var classNameTextSpan))
```

Comment:
Minor but this method could exit earlier if the original `typeName` were empty. No reason to get the `string` content of an empty span.

Reply:
I didn't want to affect the functionality of this code, as there are some subtle behaviors here. Let me know if you think it's worth pursuing.

---
In reply to: [439606336](https://github.com/dotnet/aspnetcore/pull/22887#discussion_r439606336) [](ancestors = 439606336)

Reply:
Not my code. I defer to other reviewers 😺